app/change_set_persisters/hyrax/change_set_persister.rb

```diff
@@ -32,16 +34,35 @@ def save_all(change_sets:)
       end
     end
 
-    def buffer_into_index
-      metadata_adapter.persister.buffer_into_index do |buffered_adapter|
-        with(metadata_adapter: buffered_adapter) do |buffered_changeset_persister|
-          yield(buffered_changeset_persister)
-        end
+    def delete_all(change_sets:)
+      change_sets.map do |change_set|
+        delete(change_set: change_set)
       end
     end
 
-    def with(metadata_adapter:)
-      yield self.class.new(metadata_adapter: metadata_adapter, storage_adapter: storage_adapter)
+    def delete_self_and_members(change_set:)
+      ids_to_delete = get_ids_of_self_and_all_members(resource: change_set.resource)
```

Comment:
Why not just `get_ids_of_members` as we throw away the self anyway?

Reply:
@jcoyne By default I'd prefer the function return self and members, since I think for other use cases where I just want an array of all Valkyrie::IDs in play.  In this case we're only shifting the self ID off due to the actor we're replacing having a specific code comment about deleting self last to avoid an issue with solr indexing.  In other cases where I'm using the array of ids to figure out what to load into a buffer or whatever, I'd want self in that array.